bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -81,15 +81,29 @@ async def apply_infraction(
         ctx: Context,
         infraction: _utils.Infraction,
         user: UserSnowflake,
-        action_coro: t.Optional[t.Awaitable] = None
-    ) -> None:
-        """Apply an infraction to the user, log the infraction, and optionally notify the user."""
+        action_coro: t.Optional[t.Awaitable] = None,
+        reason_override: t.Optional[str] = None,
+        additional_info: t.Optional[str] = None,
+    ) -> bool:
+        """
+        Apply an infraction to the user, log the infraction, and optionally notify the user.
+
+        `reason_override`, if provided, will be sent to the user in place of the infraction reason.
+        `additional_info` will be attached to the text field in the mod-log embed.
+        Returns whether or not the infraction succeeded.
+        """
         infr_type = infraction["type"]
         icon = _utils.INFRACTION_ICONS[infr_type][0]
         reason = infraction["reason"]
         expiry = time.format_infraction_with_duration(infraction["expires_at"])
         id_ = infraction['id']
 
+        if reason_override is not None:
+            reason_override = reason
+
+        if additional_info is not None:
+            additional_info = ""
```

Comment:
Logically this does not look right to me. How will any `reason_override` or `additional_info` show anywhere?

Additionally, an empty string seems like a better default for `additional_info` than `None` due to the way it's used.

Reply:
Shouldn't this be checking if `reason_override` or `additional_info` are `None`? Otherwise it will send `None` to the user.

Reply:
I bad :(

Resolved in 54a0ad2